src/msg/async/AsyncMessenger.cc

```diff
@@ -273,6 +277,69 @@ class C_handle_reap : public EventCallback {
   }
 };
 
+/*******************
+ * Admin Socket Hook
+ */
+
+AsyncMessengerSocketHook::AsyncMessengerSocketHook(
+    AsyncMessenger& m, const std::string& name)
+    : m_msgrs{{name, m}} {}
+
+int AsyncMessengerSocketHook::call(
+    std::string_view command, const cmdmap_t& cmdmap, const bufferlist&,
+    Formatter* f, std::ostream& errss, ceph::buffer::list& out) {
+  if (command == "messenger dump") {
+    std::string name;
+    if (common::cmd_getval(cmdmap, "msgr", name)) {
+      if (m_msgrs.contains(name)) {
+        std::vector<std::string> opts;
+        const bool tcp_info =
+            common::cmd_getval_or<bool>(cmdmap, "tcp_info", false);
+        const bool has_filter =
+            common::cmd_getval(cmdmap, "dumpcontents", opts);
+        const std::set<std::string> optset(opts.begin(), opts.end());
+        const bool all = !has_filter || optset.contains("all");
+        const auto filter_fn = [&](const std::string& key) {
+          if (key == "tcp_info") {
+            return tcp_info;
+          } else if (all) {
+            return true;
+          } else {
+            return optset.contains(key);
+          }
+        };
+
+        const auto& msgr = m_msgrs.at(name);
+        f->open_object_section("status");
+        f->open_object_section("messenger");
+        f->open_object_section(name);
+        msgr.dump(f, filter_fn);
+        f->close_section();
+        f->close_section();
+        f->close_section();
```

Comment:
```suggestion
        f->open_object_section("status");
        f->dump_string("name", name);
        f->open_object_section("messenger");
        msgr.dump(f, filter_fn);
        f->close_section();
        f->close_section();
```

seems simpler, no? (I also do not like using dynamic names for keys in JSON; it makes queries more difficult.)